### ISSUE TITLE
running example go app is broken

### ISSUE THREAD

**Issue by mistermoe** [TBD54566975/ftl#1973]
# Repro
```bash
cd examples/go
ftl dev --log-level=debug --log-timestamps
```

# Output
```
01:03:15 debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.AdminService/Ping
01:03:15 debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Ping
01:03:15 debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Ping
01:03:15 info: Starting FTL with 1 controller(s)
01:03:15 debug: Reusing existing docker container ftl-db-1 on port 15432 for postgres db
01:03:15 debug: Waiting for ftl-db-1 to be healthy
01:03:15 debug: Postgres DSN: postgres://postgres:secret@localhost:15432/ftl?sslmode=disable
01:03:15 debug: Starting DB listener
01:03:15 debug: Listening to channel: deployments_events
01:03:15 debug: Listening to channel: topics_events
01:03:15 debug:controller0: Starting FTL controller
01:03:15 debug:controller0: Building console...
01:03:15 debug: Listening to channel: topic_events_events
ftl: error: controller failed to start: context canceled
```

# Workaround
```bash
ftl dev --log-level=debug --log-timestamps --no-console
```

# Other stuff
* the first time i ran `ftl dev`, seemingly a bunch of kaka got `npm install`'d as root (no i didn't use sudo and no i'm not running as root). had to cd into frontend, nuke `node_modules` and re-run `npm install`

**Comment by alecthomas** [comment]
The root stuff is definitely an issue with your local machine (somehow), we don't do anything special with root...

**Comment by alecthomas** [comment]
I see this too, but with slightly more information:

```
🐚 ~/dev/ftl $ ftl dev --log-level=debug --recreate                                                                                          main
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.AdminService/Ping
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Ping
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Ping
info: Starting FTL with 1 controller(s)
debug: Reusing existing docker container ftl-db-1 on port 15432 for postgres db
debug: Waiting for ftl-db-1 to be healthy
debug: Postgres DSN: postgres://postgres:secret@localhost:15432/ftl?sslmode=disable
debug:migrate: Applying: 001_init.sql
debug:migrate: Applied: 001_init.sql in 110.616625ms
debug: Starting DB listener
debug: Listening to channel: deployments_events
debug: Listening to channel: topics_events
debug: Listening to channel: topic_events_events
debug:controller0: Starting FTL controller
debug:controller0: Building console...
debug:controller0: cd frontend && npm install
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug:controller0: npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
debug:controller0: npm WARN deprecated @humanwhocodes/config-array@0.11.14: Use @eslint/config-array instead
debug:controller0: npm WARN deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
debug:controller0: npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
debug:controller0: npm WARN deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
debug:controller0: npm WARN deprecated json-schema-ref-parser@6.1.0: Please switch to @apidevtools/json-schema-ref-parser
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug:controller0: 
debug:controller0: added 653 packages, and audited 654 packages in 2s
debug:controller0: 
debug:controller0: 173 packages are looking for funding
debug:controller0:   run `npm fund` for details
debug:controller0: 
debug:controller0: found 0 vulnerabilities
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug:controller0: cd frontend && npm run build
debug:controller0: 
debug:controller0: > console@0.0.0 build
debug:controller0: > tsc && vite build
debug:controller0: 
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug:controller0: vite v4.5.3 building for production...
debug:controller0: transforming...
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug: Unary RPC failed: unavailable: dial tcp [::1]:8892: connect: connection refused: /xyz.block.ftl.v1.ControllerService/Status
debug:controller0: ✓ 1756 modules transformed.
debug: Unary RPC failed: deadline_exceeded: context deadline exceeded: /xyz.block.ftl.v1.ControllerService/Status
error: Timeout reached while polling for controller status: context deadline exceeded
error: Failed to receive notification: context canceled
ftl: error: controller failed to start: context deadline exceeded
🐚 ~/dev/ftl $                                                                                                                               main
```

**Comment by alecthomas** [comment]
Possibly a duplicate of #1942

**Comment by alecthomas** [comment]
Repro is pretty simple:

```
git clean -fdx frontend
ftl dev --recreate --log-level=debug
```

It looks like (somehow) we've introduced a context timeout somewhere. By forcing a rebuild of the frontend in dev mode, it triggers pretty reliably.

**Comment by gak** [comment]
Fixed in #1979
